ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/HiveTypeSystemImpl.java

```diff
@@ -148,4 +172,17 @@ public boolean isSchemaCaseSensitive() {
     return false;
   }
 
+  @Override
+  public RelDataType deriveSumType(RelDataTypeFactory typeFactory,
```

Comment:
Can you add a comment explaining why is this being done? Adding 10 seems arbitrary without the comment.